setuptools/package_index.py

```diff
@@ -588,24 +588,26 @@ def find(req, env=None):
                     continue
 
                 if dist in req and (dist.precedence<=SOURCE_DIST or not source):
-                    return dist
+                    mylocation = self.download(dist.location, tmpdir)
+                    if os.path.exists(mylocation):
+                        return dist, mylocation
 
         if force_scan:
             self.prescan()
             self.find_packages(requirement)
-            dist = find(requirement)
+            dist, mylocation = find(requirement)
 
```

Comment:
Since you are modifying this chunk...maybe?

   if not dist and local_index is not None: